pkg/engine/task/task.go

```diff
@@ -189,17 +189,17 @@ func fatalExecutionError(cause error, eventName string, meta renderer.Metadata)
 
 func newKudoOperator(task *v1beta1.Task) (Tasker, error) {
 	// validate KudoOperatorTask
-	if len(task.Spec.KudoOperatorTaskSpec.Package) == 0 {
+	if task.Spec.KudoOperatorTaskSpec.Package == "" {
 		return nil, fmt.Errorf("task validation error: kudo operator task '%s' has an empty package name", task.Name)
 	}
 
-	if len(task.Spec.KudoOperatorTaskSpec.OperatorVersion) == 0 {
+	if task.Spec.KudoOperatorTaskSpec.OperatorVersion == "" {
 		return nil, fmt.Errorf("task validation error: kudo operator task '%s' has an empty operatorVersion", task.Name)
 	}
 
 	return KudoOperatorTask{
 		Name:            task.Name,
-		Package:         task.Spec.KudoOperatorTaskSpec.Package,
+		OperatorName:    task.Spec.KudoOperatorTaskSpec.Package,
```

Comment:
We should rename this in `KudoOperatorTaskSpec` as well. As that API isn't in use yet it's safe to rename without breaking backwards-compatibility.

Reply:
Actually, `KudoOperatorTaskSpec` needs both:
- `Package` is set by the operator developer which KUDO will try to resolve as usual (local, URL, community repo)
- `OperatorName` is the result of resolving the package, set by the CLI used by the manager

See the commit below

Reply:
:+1: That should work!